src/components/Rewards/withdraw.tsx

```diff
@@ -0,0 +1,225 @@
+  display: flex;
+  padding-left: 24px;
+  padding-right: 24px;
+  flex: wrap;
+  padding-bottom: 24px;
+  margin: auto;
+  width: 100%;
+  overflow: hidden;
+  text-align: left;
+  justify-content: flex-end;
+`
+const InputWrapper = styled('div')`
+  display: flex;
+  flex: wrap;
+  margin: auto;
+  border-radius: 12px;
+  padding: 12px;
+  border: 2px solid white;
+  width: 100%;
+  overflow: hidden;
+  text-align: left;
+  justify-content: flex-end;
+  > span {
+    margin: auto;
+  }
+`
+const Input = styled('input')`
+  display: flex;
+  flex: 1;
+  font-size: 16px;
+  background: none;
+  border: none;
+  padding: 0.5rem;
+  color: white;
+  :focus {
+    outline: none;
+  }
+`
+const Text = styled('div')`
+  display: flex;
+  justify-content: flex-end;
+  padding-right: 10px;
+  text-align: right;
+  color: #b5b9d3;
+  font-size: 14px;
+  font-weight: 400;
+  line-height: 18px;
+`
+
+const Balance = styled('div')`
+  display: flex;
+  text-align: center;
+  font-size: 14px;
+  font-weight: 500;
+  line-height: 18px;
+`
+
+const Percentage = styled('button')`
+  margin: 2px;
+  margin-top: 10px;
+  padding: 4px;
+  display: inline-block;
+  text-align: center;
+  justify-content: flex-end;
+  color: white;
+  font-size: 14px;
+  font-weight: 400;
+  line-height: 18px;
+  width: 100%;
+  border-radius: 999px;
+  border: solid 2px white;
+  background: none;
+`
+
+interface RewardProps {
+  withdrawValue?: number
+  data: {
+    lpBalance: string
+    rewardUnlockedUser: string
+    rewardEstimate: string
+    rewardTotal: string
+    rewardUnlocked: string
+  }
+  contract: {
+    stakingContractAddress: string
+    tokenAddress: string
+    user: string
+    token0: string
+    token1: string
+  }
+}
+
+export default function WithdrawReward(props: RewardProps) {
+  const [withdrawValue, setWithdrawValue] = useState(props.withdrawValue)
+
+  const [contract, setContract] = useState<{
+    stakingContractAddress: string
+    tokenAddress: string
+    user: string
+    token0: string
+    token1: string
+  }>({ stakingContractAddress: '', tokenAddress: '', user: '', token0: '', token1: '' })
+  const [result, setResult] = useState<{
+    lpBalance: string
+    rewardUnlockedUser: string
+    rewardEstimate: string
+    rewardTotal: string
+    rewardUnlocked: string
+  }>({ lpBalance: '0', rewardUnlockedUser: '0', rewardEstimate: '0', rewardTotal: '0', rewardUnlocked: '0' })
+  const [estimate, setEstimate] = useState('0')
+
+  const { library } = useActiveWeb3React()
+
+  function selectPercentage(amount: number) {
+    const calculated = (Number(result.lpBalance) * amount) / 100
+    const rewards = Number(result.rewardEstimate) - (Number(result.rewardEstimate) * amount) / 100
+    setEstimate(rewards.toFixed(2).toString())
+    setWithdrawValue(calculated)
+  }
+
+  useEffect(() => {
+    setEstimate(props.data.rewardEstimate)
+    setWithdrawValue(0)
+    setContract(props.contract)
+    setResult(props.data)
+  }, [props])
```

Comment:
better to de structure this so that the component re-renders only when certain properties change

Reply:
let's skip that one for now